deployment/7_upgrade_v2.js

```diff
@@ -0,0 +1,133 @@
+const DeployManager = require('../utils/deploy-manager.js');
+const MultisigExecutor = require('../utils/multisigexecutor.js');
+const semver = require('semver');
+
+const TARGET_VERSION = "1.1.0";
+const MODULES_TO_ENABLE = ["NftTransfer"];
+const MODULES_TO_DISABLE = [];
+const BACKWARD_COMPATIBILITY = 1;
+
+const deploy = async (network) => {
+
+    const newModuleWrappers = [];
+    const newVersion = {};
+
+    ////////////////////////////////////
+    // Setup
+    ////////////////////////////////////
+
+    const manager = new DeployManager(network);
+	await manager.setup();
+
+	const configurator = manager.configurator;
+	const deployer = manager.deployer;
+    const abiUploader = manager.abiUploader;
+    const versionUploader = manager.versionUploader;
+    const deploymentWallet = deployer.signer;
+    const config = configurator.config;
+
+    const ModuleRegistryWrapper = await deployer.wrapDeployedContract(ModuleRegistry, config.contracts.ModuleRegistry);
+    const MultiSigWrapper = await deployer.wrapDeployedContract(MultiSig, config.contracts.MultiSigWallet);
+    const multisigExecutor = new MultisigExecutor(MultiSigWrapper, deploymentWallet, config.multisig.autosign);
+    
+    console.log('Config:', config);
+
+    ////////////////////////////////////
+    // Deploy new modules
+    ////////////////////////////////////
+
+    const NftTransferWrapper = await deployer.deploy(
+        NftTransfer,
+        {},
+        config.contracts.ModuleRegistry,
+        config.modules.GuardianStorage
+    );
+    newModuleWrappers.push(NftTransferWrapper);  
+    
+    ///////////////////////////////////////////////////
+    // Update config and Upload new module ABIs
+    ///////////////////////////////////////////////////
+
+    configurator.updateModuleAddresses({
+        NftTransfer: NftTransferWrapper.contractAddress
+    });
+
+    const gitHash = require('child_process').execSync('git rev-parse HEAD').toString('utf8').replace(/\n$/, '');
+    configurator.updateGitHash(gitHash);
+    await configurator.save();
+
+    await Promise.all([
+        abiUploader.upload(NftTransferWrapper, "modules")
+    ]);
+
+    ////////////////////////////////////
+    // Register new modules
+    ////////////////////////////////////
+
+    for (let idx = 0; idx < newModuleWrappers.length; idx++) {
+        let wrapper = newModuleWrappers[idx];
+        await multisigExecutor.executeCall(ModuleRegistryWrapper, "registerModule", [wrapper.contractAddress, utils.asciiToBytes32(wrapper._contract.contractName)]);
+    }
+
+    ////////////////////////////////////
+    // Deploy and Register upgraders
+    ////////////////////////////////////
+
+    const toAdd = newModuleWrappers.map((wrapper) => {
+        return {
+            'name': wrapper._contract.contractName,
+            'address': wrapper.contractAddress
+        };
+    });
+    let fingerprint; 
+
+    const versions = await versionUploader.load(BACKWARD_COMPATIBILITY);
```

Comment:
What exactly does this load?

Reply:
It loads the Version object from the `BACKWARD_COMPATIBILITY` previous versions. We need them to construct and deploy the Upgrader contract that will upgrade wallets from a past version (within `BACKWARD_COMPATIBILITY` range) to the target version.